### PR TITLE
PP-878: pbs_swigify is broken

### PR DESCRIPTION
**Pull request by sujatapatnaik52**
#### Issue-ID

* **[PP-878](https://pbspro.atlassian.net/browse/PP-878)**

#### Problem description
* pbs_swigify was broken. Thus, unable to use API mode in PTL.

#### Cause / Analysis
* Problem with Linking archived library libpbs.a using -fPIC.

#### Solution description
* Generating the _pbs_ifl.so by dynamically linking the libpbs.so with -fPIC.

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
- [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [ ] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
- [ ] I have added  **manual test(s) to the Jira ticket and explained why PTL is not appropriate** for this case.
- [ ] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [x] I have attached **test logs to the Jira ticket** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

File: test/fw/bin/pbs_swigify [hunk 1]
```diff
@@ -305,9 +305,11 @@ if __name__ == '__main__':
         cmd = ['gcc', '-Wall', '-Wno-unused-variable', '-fPIC', '-shared',
                '-I' + pbsinclude]
         cmd += ['-I' + pythoninc]
-        cmd += ['pbs_ifl_wrap.c', pbs_conf['PBS_EXEC'] + '/lib/libpbs.a']
-        cmd += ['-lcrypto']
+        cmd += ['pbs_ifl_wrap.c', '-L' +
+                pbs_conf['PBS_EXEC'] + '/lib', '-lpbs']
```
Comment (hirenvadalia) [inline test/fw/bin/pbs_swigify]: Please this and above line with below:
```
cmd += ['pbs_ifl_wrap.c', '-L']
cmd += [os.path.join(pbs_conf['PBS_EXEC'], 'lib')]
cmd += ['-lpbs']
```

File: test/fw/bin/pbs_swigify [hunk 2]
```diff
@@ -305,9 +305,11 @@ if __name__ == '__main__':
...
         cmd += ['-o', '_pbs_ifl.so']
+        cmd += ['-L' + '/lib']
```
Comment (hirenvadalia) [inline test/fw/bin/pbs_swigify]: Do we need this? I thought gcc at-least looks into /lib (and/or /lib64) directory for libraries...
Can we try out removing this line and see whether it works or not?

Comment (sujatapatnaik52) [inline test/fw/bin/pbs_swigify]: I tried removing the library path and it's working as expected. I can remove it.

**Comment by agrawalravi90** [comment]
Thanks for the changes Sujata. According to our checklist, as you are not checking in a PTL test with this PR, could you please comment on the ticket explaining why a PTL test is not feasible/appropriate for this change?
Also, and this is optional, from our discussion yesterday, it sounded like this ticket will not completely fix PTL's API mode, so, again optional, but if there are other tickets filed for further fixing the API mode, could you please link them to this ticket?

Thanks!

**Comment by sujatapatnaik52** [comment]
@agrawalravi90 This fix is mainly to generate the _pbs_ifl.so file and verify that can be used further in the API mode. 
According to the description of this ticket - https://pbspro.atlassian.net/browse/PP-889, there are some bugs which needs to be fixed in the API mode.

**Comment by agrawalravi90** [comment]
Ok, works for me, please squash and rebase

**Pull request merged**